project-addons/product_weight_net/models/product.py

```diff
@@ -30,6 +30,11 @@ def write(self, values):
             boms = self.env['mrp.bom'].search([('bom_line_ids.product_id', '=', self.id)])
             for product in boms.mapped('product_tmpl_id'):
                 product.product_variant_ids.calculate_bom_weight()
+
+        if values.get('volume', False) and self.used_in_bom_count:
+            boms = self.env['mrp.bom'].search([('bom_line_ids.product_id', '=', self.id)])
```

Comment:
aquí pasa lo mismo, si se hace una actualización al tiempo de varios productos fallaría, lo mismo pasa justo arriba en el código que ya había. Supongo que podrías hacer, en ambos sitios: boms = self.env['mrp.bom'].search([('bom_line_ids.product_id', 'in', self.ids)]) en este caso y evitas el for x in self